File: src/main/java/com/cpjd/speechGeneration/SilenceAudioSendHandler.java
package com.cpjd.speechGeneration;

import net.dv8tion.jda.core.audio.AudioSendHandler;

public class SilenceAudioSendHandler implements AudioSendHandler {
    private static final byte[] silenceBytes = new byte[] {(byte)0xF8, (byte)0xFF, (byte)0xFE};

    private boolean canProvide = true;
    private long startTime = System.currentTimeMillis();

    @Override
    public byte[] provide20MsAudio() {

        // send the silence only for 5 seconds
        if(((System.currentTimeMillis() - startTime) / 1000) > 5) {
            canProvide = false;
        }

        return silenceBytes;
    }

    @Override
    public boolean canProvide() {
        return canProvide;
    }

    @Override
    public boolean isOpus() {
        return true;
    }
}

File: src/main/java/com/cpjd/audio/SilenceAudioSendHandler.kt
package com.cpjd.audio

import net.dv8tion.jda.core.audio.AudioSendHandler
import java.util.*

class SilenceAudioSendHandler: AudioSendHandler {
    var canProvide = true
    var startTime = Date().time

    override fun provide20MsAudio(): ByteArray {

        // send the silence only for 5 seconds
        if (((Date().time - startTime) / 1000) > 5) {
            canProvide = false
        }

        val silence = arrayOf(0xF8.toByte(), 0xFF.toByte(), 0xFE.toByte())
        return silence.toByteArray()
    }

    override fun canProvide(): Boolean {
        return canProvide
    }

    override fun isOpus(): Boolean {
        return true
    }
}